packages/core/core/src/services/document-service/transform/relations/transform/data-ids.ts

```diff
@@ -70,39 +73,47 @@ const transformDataIdsVisitor = (idMap: IdMap, data: Record<string, any>, source
       if (!attribute) {
         return;
       }
-
-      // Find relational attributes, and return the document ids
-      const targetUid = attribute.target!;
-      const getIds = getRelationIds(idMap, targetUid, source);
+      const isPolymorphic = isPolymorphicRelation(attribute);
+      const getIds = getRelationIds(idMap, source);
 
       // Transform the relation documentId to entity id
       const newRelation = await mapRelation((relation) => {
         if (!relation || !relation.documentId) {
           return relation;
         }
 
-        const ids = getIds(relation);
+        // Find relational attributes, and return the document ids
+        // if its a polymorphic relation we need to get it from the data itself
+        const targetUid = (attribute.target || relation.__type) as UID.Schema;
+        const ids = getIds(targetUid, relation);
 
         // Handle positional arguments
         const position = { ...relation.position };
 
         if (position.before) {
           const beforeRelation = { ...relation, ...position, documentId: position.before };
-          position.before = getIds(beforeRelation).at(0);
+          // TODO: Reordering on polymorphic relations
```

Comment:
Might be silly, but could you be a bit more precise on what's left to do regarding the reordering (or why isn't it working yet)?

Reply:
I think this can be removed actually. @Marc-Roig do you know what would still be left here if anything? All the tests for re-ordering are implemented and passing I believe

Reply:
this is still the case, we could make it work instead of keeping the todo

if connecting before / after a relation (polymorphic), we should also specify the __type uid of the specified before/after relation, atm we are using the uid of the relation being reordered.

e.g
```ts
{
  "connect": {
    "documentId": "X",
    "__type": "api::category.category",
    "position": {
      "before": {
        "documentId": "Y", 
        "type": "api::article.article" // This is what we are not using here
      }
    }
  }
}```

Reply:
Ahhh, so the reason the tests are passing is because the uids happen to be the same, that makes sense.

It seems like we might not need it at all though, the document id alone should be enough to find it in the relations table. We can take a look tomorrow